flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -1400,6 +1401,25 @@ public static String formatUnixTimestamp(long unixtime, TimeZone tz) {
         return formatUnixTimestamp(unixtime, TIMESTAMP_FORMAT_STRING, tz);
     }
 
+    /** Add offset to string (like '1970-01-01') and return new date For String support. */
+    public static Integer dateAdd(BinaryStringData binaryStringData, int offset) {
+
+        if (binaryStringData == null) {
+            return null;
+        }
+        try {
+            int days = parseDate(binaryStringData.toString());
+            return days + offset;
+        } catch (NullPointerException e) {
```

Comment:
Isn't it possible to prevent NullPointerException with some extrachecks rather than catching it?

Reply:
i modify this, add execption log. just reference formatUnixTimestamp，if string args is not  legal format,log and return null.